pkg/scheduler/scheduler.go

```diff
@@ -263,7 +267,7 @@ func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 			return
 		}
 		log.Error(err, errCouldNotAdmitWL)
-		s.requeueAndUpdate(log, ctx, &e.Info, err.Error())
+		s.requeueAndUpdate(log, ctx, *e)
 		s.recorder.Eventf(newWorkload, corev1.EventTypeNormal, "Pending", "%s: %v", errCouldNotAdmitWL, err)
```

Comment:
we shouldn't post an event if the workload was skipped, same thing with status update below.

Reply:
I think we can bring the Event recording inside `requeueAndUpdate` (and skip it accordingly)

Reply:
Yes. Moved them into `requeueAndUpdate `